File: 03-architecture/imooc-music-app-refactor/ft_login/src/main/java/com/imooc/ft_login/presenter/UserLoginPresenter.java
package com.imooc.ft_login.presenter;

import com.google.gson.Gson;
import com.imooc.ft_login.api.MockData;
import com.imooc.ft_login.api.RequestCenter;
import com.imooc.ft_login.inter.IUserLoginPresenter;
import com.imooc.ft_login.inter.IUserLoginView;
import com.imooc.ft_login.manager.UserManager;
import com.imooc.lib_base.ft_login.model.LoginEvent;
import com.imooc.lib_base.ft_login.model.user.User;
import com.imooc.lib_network.okhttp.listener.DisposeDataListener;
import org.greenrobot.eventbus.EventBus;

/**
 * 登陆页面对应Presenter
 */
public class UserLoginPresenter implements IUserLoginPresenter, DisposeDataListener {

    private IUserLoginView mIView;

    public UserLoginPresenter(IUserLoginView iView) {
        mIView = iView;
    }

    @Override
    public void login(String username, String password) {
        mIView.showLoadingView();
        RequestCenter.login( this);
    }

    @Override
    public void onSuccess(Object responseObj) {
        mIView.hideLoadingView();
        User user = (User) responseObj;
        UserManager.getInstance().setUser(user);
        //发送登陆Event
        EventBus.getDefault().post(new LoginEvent());
        mIView.finishActivity();
    }

    @Override
    public void onFailure(Object reasonObj) {
        mIView.hideLoadingView();
        onSuccess(new Gson().fromJson(MockData.LOGIN_DATA, User.class));
        mIView.showLoginFailedView();
    }
}

File: 03-architecture/imooc-music-app/imooc_voice/src/main/java/com/imooc/imooc_voice/view/login/presenter/UserLoginPresenter.kt
package com.imooc.imooc_voice.view.login.presenter

import com.google.gson.Gson
import com.imooc.imooc_voice.api.KRequestCenter
import com.imooc.imooc_voice.api.MockData
import com.imooc.imooc_voice.model.login.LoginEvent
import com.imooc.imooc_voice.model.user.User
import com.imooc.imooc_voice.utils.UserManager
import com.imooc.imooc_voice.view.login.inter.IUserLoginPresenter
import com.imooc.imooc_voice.view.login.inter.IUserLoginView
import com.imooc.lib_network.okhttp.listener.DisposeDataListener
import com.imooc.lib_network.okhttp.utils.ResponseEntityToModule
import org.greenrobot.eventbus.EventBus

class UserLoginPresenter constructor(userLoginView: IUserLoginView) : IUserLoginPresenter, DisposeDataListener {

    private val mUserLoginView: IUserLoginView = userLoginView

    override fun login(name: String, password: String) {
        mUserLoginView.showLoadingView()
        KRequestCenter.login(this)
    }

    override fun onSuccess(responseObj: Any) {
        mUserLoginView.hideLoadingView()
        //保存到单例对象中
        UserManager.mUser = (responseObj as User)
        //发送登陆Event
        EventBus.getDefault().post(LoginEvent())
        mUserLoginView.finishActivity()
    }

    override fun onFailure(reasonObj: Any) {
        mUserLoginView.hideLoadingView()
        //onSuccess(ResponseEntityToModule.parseJsonToModule(MockData.LOGIN_DATA, User::class.java))
        onSuccess(Gson().fromJson(MockData.LOGIN_DATA, User::class.java))
        mUserLoginView.showLoginFailedView()
    }
}